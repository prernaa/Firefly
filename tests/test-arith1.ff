x = 1 + 2
y = 3.1 - 0.0
z = -1.0 * 2.5
p = 5.0 / 2.0