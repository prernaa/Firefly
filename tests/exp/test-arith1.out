
	#include <iostream>
#ifdef _WIN32
    #include <windows.h>
#endif
#include <GL/glut.h>  // GLUT, includes glu.h and gl.h
#include <math.h>
#include <stack>
#ifdef _MSC_VER                         // Check if MS Visual C compiler
#  pragma comment(lib, "opengl32.lib")  // Compiler-specific directive to avoid manually configuration
#  pragma comment(lib, "glu32.lib")     // Link libraries
#endif	
	
using namespace std;
struct vec2cpp{
	float x;
	float y;
	vec2cpp& operator=(const vec2cpp& other)
	{
		x = other.x;
		y = other.y;
		return *this;
	}
	vec2cpp& operator+(const vec2cpp& other)
	{
		x += other.x;
		y += other.y;
		return *this;
	}
	vec2cpp& operator-(const vec2cpp& other)
	{
		x -= other.x;
		y -= other.y;
		return *this;
	}
	vec2cpp& operator*(const vec2cpp& other)
	{
		x *= other.x;
		y *= other.y;
		return *this;
	}
	vec2cpp& operator/(const vec2cpp& other)
	{
		x /= other.x;
		y /= other.y;
		return *this;
	}
	bool operator==(const vec2cpp& other)
	{
		return (x==other.x && y==other.y);
	}
	bool operator!=(const vec2cpp& other)
	{
		return !(x==other.x && y==other.y);
	}
};
struct actRecord{
	void *retFunc;
};
void* returnPath;
int myprogram();
void DrawAxes()
{
	glBegin(GL_LINES);
		glColor3f(1.0, 0.0, 0.0);
		glVertex2f(-1.0f, 0.0f);
		glVertex2f(+1.0f, 0.0f);
		glColor3f(0.0, 0.0, 1.0);
		glVertex2f(0.0f, -1.0f);
		glVertex2f(0.0f, +1.0f);
	glEnd();
}
void display() {
	glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT); // Clear color and depth buffers
   	glMatrixMode(GL_MODELVIEW);     // To operate on model-view matrix

	// Render a color-cube consisting of 6 quads with different colors
   	glLoadIdentity();                 // Reset the model-view matrix

   	DrawAxes();
	myprogram();

  	glutSwapBuffers();  // Swap the front and back frame buffers (double buffering)
}
void init() {
	glClearColor(1.0, 1.0, 1.0, 1.0);
	glColor3f(0.0, 0.0, 0.0);
   	glClearDepth(1.0f);                   // Set background depth to farthest
   	glEnable(GL_DEPTH_TEST);   // Enable depth testing for z-culling
   	glDepthFunc(GL_LEQUAL);    // Set the type of depth-test
   	glShadeModel(GL_SMOOTH);   // Enable smooth shading
   	glHint(GL_PERSPECTIVE_CORRECTION_HINT, GL_NICEST);  // Nice perspective corrections
}
void reshape(GLsizei width, GLsizei height) {  // GLsizei for non-negative integer
// Compute aspect ratio of the new window
   if (height == 0) height = 1;                // To prevent divide by 0
   GLfloat aspect = (GLfloat)width / (GLfloat)height;
 
   // Set the viewport to cover the new window
   glViewport(0, 0, width, height);
 
   // Set the aspect ratio of the clipping area to match the viewport
   glMatrixMode(GL_PROJECTION);  // To operate on the Projection matrix
   glLoadIdentity();
   if (width >= height) {
     // aspect >= 1, set the height from -1 to 1, with larger width
      gluOrtho2D(-1.0 * aspect, 1.0 * aspect, -1.0, 1.0);
   } else {
      // aspect < 1, set the width to -1 to 1, with larger height
     gluOrtho2D(-1.0, 1.0, -1.0 / aspect, 1.0 / aspect);
   }
}
int main(int argc, char** argv)
{
	glutInit(&argc, argv);          // Initialize GLUT
	glutInitDisplayMode(GLUT_DEPTH | GLUT_DOUBLE | GLUT_RGBA);	// Enable double buffering
   	glutInitWindowSize(640, 480);   // Set the window's initial width & height - non-square
   	glutInitWindowPosition(50, 50); // Position the window's initial top-left corner
   	glutCreateWindow("Model Transform");  // Create window with the given title
   	glutDisplayFunc(display);       // Register callback handler for window re-paint event
   	glutReshapeFunc(reshape);       // Register callback handler for window re-size event
   	init();                       // Our own OpenGL initialization
   	glutMainLoop();                 // Enter the infinite event-processing loop
	return 0;
}
int myprogram(){
	vec2cpp _ff = {0,0};
	stack <actRecord> fRecords;
	float lclVars[1024];
	int lv_index = 0;
	stack <int> lv_frameptr;
	stack <int> lv_lastframeptr;
	
	int x;
	float y;
	float z;
	float p;
	int _t0;
	int _t1;
	int _t2;
	float _t3;
	float _t4;
	float _t5;
	float _t6;
	float _t7;
	float _t8;
	float _t9;
	float _t10;
	float _t11;

	_t0 = 1;
	_t1 = 2;
	_t2 = _t0 + _t1;
	x = _t2;
	_t3 = 3.1;
	_t4 = 0.;
	_t5 = _t3 - _t4;
	y = _t5;
	_t6 = -1.;
	_t7 = 2.5;
	_t8 = _t6 * _t7;
	z = _t8;
	_t9 = 5.;
	_t10 = 2.;
	_t11 = _t9 / _t10;
	p = _t11;


	return 0;
}
